### PR TITLE
Archery polish/cleanup/rebalance first pass

### PR DESCRIPTION
**Pull request by Cosmitz**
This PR deals with the archery branch and cleaning it up as well as redressing crossbows properly to rifles after their removal. Below the current changes, at the bottom you'll find a breakdown of the methodology of design applied for the rework.

Crossbows related:

- Moved bolt fabrication to marksmanship from archery

- Moved crossbows and bolts to ranged.json given their move to rifles

- Kept crowssbows in the archery item group --- may need changing? will just make it hard for archery focused characters to find archery supplies as a crossbow are now useless for them

Bows and arrows:

- Removed old crafting components (arrowheads/shafts) and items which are no longer valid/obtainable

- Re-allowed crafting the exploding arrow and gave it a simple craft recipe like the rest of the current arrows

- Reintroduced flaming arrow via crafting and balanced, it's a relic but it's nifty, it starts out crafted as an unshootable arrow and atm is bugged when fired to stay 'flaming' in the inventory, not fixed for this pass, also moved recipe for flamable arrow into archery from tools

- Added wood saw requirements to crafting short/longbow as well as more materials and synchronised to the progression in the archery skill. Also added CUT FINE requirement instead of CUT.

- Removed the reflex-recurve which isn't a thing, blended it into the composite-reflex, bows remaining: (self, short, composite-reflex, recurve, compound, longbow) - any extravagant bows or 'hun horse archery bow circa 1850' can be modded in

- Added fine cutting 1 to pocket knife, it was had by a lot bigger knives and makes sense to be had by a pocket knife (also fine cutting now used for crafting arrows)

- Reworked requirements on crafting arrows from former 1mat-1arrow to require roughly double or triple mats for the set of 10 crafted, this offsets bow's slightly increased efficiency

- Lowered intelligence to 5 for novice bowhunter book

- Fixed item groups spawns and static spawns with new IDs and balance-focused. The craftables short/longbows come out less since they were not common in pre-world, but the uncraftable bows, compound/recurve and composite have a fair chance of dropping, also equalised chance with crossbows

- Reworked crafting requirements for bow stabilizer and arrowrest (they were 1 plastic scrap with no tools) as well as rebalanced their dispersion bonuses

- Rebalanced bows/arrows crafting materials. Some recipies required old '1 item of each' for the new 10-count arrows. Others didn't but didn't use the proper recipies. Unified the concept of what items are required. Craftables bows are now: Self/Short/Longbow. New stats are balance by a lot more crafting materials needed for crafting a set of 10 arrows or 5 explosive/flamable arrows instead of the very cheap options now availible which make arrows almost free.

- Reworked STR DRAW as using half the difference between Player strenght and minstrenght as added range to the bows, in turn increased min str requirement and lowered overall range to compensate for a roughly 10-12 str character. It was a very uneven penalty requiring extremely high strenght to get the basics out of a bow. It was also unrealistic since a properly drawn bow will give the maximum power it has, 'overdrawing' adds very minimal extra power and risks damaging the bow. In game terms, bows themselves are balanced for the range, with strenght bonuses adding on top very mildly, but with bows themselves requiring greater strenght to wield

- Made tiny change in NW mod for compatibility, only in regards to proper item IDs, no stats were changed

- Moved obsolete items into /data/json/items/obsolete_archery_compatibility.json and made sure there is no way to legally get them in the game so the file can be safely removed, it was only created to ignore errors in existing save games with those items

TO DO: 

- Proper stat rebalance which i'm having issues with given the new math/dispersion/aiming time formulas.

- Flaming arrows proper implementation

- Archery thumb ring.

---------

Methodology of design:

The player has three craftable bows he can make. The self bow is the basic bow which should be used for target practice with the soon-followed craftable Short Bow to take up the slack as a good starter weapon. Shortbows and Composite reflex bows are shorter ranged items which have a very quick reload compared to the larger bows and the final concept for them is to be the quick-steadying, slightly innacurate options for the bowman on the move. Composite reflex bows are loot-find only.

Longbows are the last craftable variant which is also the furthest-shooting bow but at a decent penalty to damage or range. It's imagined the player field-shoots said longbow. Compound bows are very strong and precise, but have a long reloading time and i intend to give them a long aim time as well with good accuracy. A marksman's bow. Recurves are the middle class of the last three bows, uncraftable andt firmly in the middle power, accuracy and range wise. 

Arrows are all raw-to-final-product recipes with all crafting relying firmly on fabrication with archery becoming a requirement later on. Unfletched arrow shafts are good just to be used as training, soon replaced with wooden arrows which require just a bit more materials. Fire-hardened wooden arrows should be the bread and butter for long range shots and arrows that can take a hit. Good for starting the midgame. Field points and heavy points are the proper damage variants, with the heavy field point doing the maximum damage even carbon/aluminum arrows do, but at a range penalty.

Copper, aluminum and carbon are the top tier arrows for end game, with carbon being the only one craftable and very brittle, as well as not as good as the found aluminum/carbon. The last two are low durability/high durability variants which should be the go-to for endgame. Drops are framed around carbon arrows being harder to find than aluminum.

Flaming and Exploding arrows are specialist tools with significant drawbacks. At least for Exploding, as flaming are currently bugged in a 'i have fire arrows on fire in my backpack' kind of way, which will need to be solved at a furter time. But atm, those work as functional items and are not OP.

--------

The STR matter.

It never made sense for bows to increase substantially in range due to strenght. That's not how bows work. However, there is some range/power that can be pulled from a bow from overdrawing. Later on i may rework this as a reload bonus, but for now, i changed the formula for STR DRAW to:

ret += std::max( 0, ( p->get_str() - type->min_str ) * 0.5 );

So if a 11 strenght player shoots a 5 strenght bow, he will get 3 extra range. A 19 strenght player shooting a 5 bow will get 8, which on top of the roughly 26 max range, will make it worth their time.

### CODE REVIEW

File: data/json/item_groups.json [hunk 1]
```diff
@@ -2158,20 +2158,22 @@
...
       ["vambrace_larmor", 1],
       ["quiver", 3],
       ["hand_crossbow", 1],
-      ["arrow_wood", 7],
-      ["arrow_cf", 5],
+      ["arrow_aluminum", 3],
+      ["arrow_cf", 3],
+	  ["arrow_wood_field", 5],
+      ["arrow_wood_field_heavy", 5],
+	  ["arrow_wood", 10],
```
Comment (Coolthulhu) [inline data/json/item_groups.json]: Replace all tabs in the PR with spaces. If you're using notepad++, it has tabs-to-spaces replacement, or you can use regex to do that.

**Comment by kevingranade** [comment]
This needs to be broken up, 1,000+ lines of misc changes is excessive and will make review take a very long time.
In particular, please break up at least:
1. Mechanical changes like removing obsolete entries and moving entries.
2. Mechanics changes, such as any skill changes or crafting requirement changes.
3. Content removal.
4. Balance changes.

**Comment by Coolthulhu** [comment]
This is a huge PR, combining multiple related but not dependent elements. It's hard to review huge PRs like that.
Split the bow part from the arrow part.

File: data/json/recipes/recipe_ammo.json [hunk 2]
```diff
@@ -1,685 +1,565 @@
...
+	  [ "bowl_plastic", 3 ],
+	  [ "plastic_shopping_bag", 2],
+	  [ "cup_plastic", 3 ],
+	  [ "jug_plastic", 2],
       [ "box_small", 1 ],
-      [ "plastic_chunk", 1 ],
-      [ "aluminum_foil", 40 ],
+      [ "plastic_chunk", 2 ],
+      [ "aluminum_foil", 30 ],
       [ "cash_card", 2 ]
```
Comment (Cosmitz) [inline data/json/recipes/recipe_ammo.json]: Like how 'cording' deals with various strings and such. Still, given these will probably only be used in this configuration for archery-related, don't know if it's very needed. But will do it if required.

File: data/json/item_groups.json (hunk 1 above)
Comment (Cosmitz) [inline data/json/item_groups.json]: Sure, will do.

**Comment by Cosmitz** [comment]
Ok will get to work splitting this down, but will leave this up as the general working document for the overall changes.

**Comment by Night-Pryanik** [comment]
> Removed the reflex-recurve which isn't a thing

This was already discussed in #13478. In short: you are wrong, they do exist.

> Moved bolt fabrication to marksmanship from archery

How marksmanship correlates to the bolt fabrication? One can be a super accurate sniper, but have no idea about crafting bolts.

Also, about reducing number of craftable bows to only three. I didn't see an explanation on why a skilled bowyer couldn't craft high-level bows.

**Comment by Cosmitz** [comment]
They exist but in the current setup were practically indistinguishable even naming wise. I'm reframing archery as a whole then will push for diversity within the classes. We really didn't need 'deflex recurve' and 'reflex decurve' when you could do so much more with the bow class than spin words people will get mixed up.

And bolts relats to marksmanship because right now you have to learn archery to build bolts which you shoot out of a crossbow that's linked to rifles. To be fair, might as well link them to fabrication with a sub-requisitory in rifles for crossbows as i've done with arrows and archery, but just wanted to give it a quick needed fix and not muck about them too much since they're not part of my scope right now. 

As for crafting, with the current infrastructure we have in the game, it'd be relatively hard to make the next one up, a recurve or even the composite. Went back and forth over it but for the moment decided on that, but that doesn't mean there won't be more craftable bows later on. I'd just like to add a few more requirements in terms of tools/workbenches for the other bow constructions which won't be a part of this initial pass. 

Either way, once i get my head around proper use of github i'll start churning out these smaller fixes.

**Comment by Leland** [comment]
> Added wood saw requirements to crafting short/longbow as well as more materials and synchronised to the progression in the archery skill. Also added CUT FINE requirement instead of CUT.

Not sure I agree with this. `CUT_FINE` refers to a level of precision required from surgical tools. I don't think whittling and fletching necessitates a greater precision of cutting than just `CUT`.  The complexity of crafting recipes is delivered through a skill level requirement rather than a quality flag on a tool.

**Comment by Cosmitz** [comment]
Cut is applied to more than enough melee weapons. If i'd use CUT i could do fletchings of soft feathers with a broadsword or a wood axe. CUT 2 is almost completely unused as well. CUT FINE is a more narrow but still acccessible category of tools and also makes more sense. 

If anything, we should be promoting more 'right tool for the job' rather than using a tanto for everything. (I still reel at how you can butcher a rabbit with something five times its size but that's for another PR regarding BUTCHERY on everything)

**Comment by kevingranade** [comment]
Obsoleted by subsequent PRs and there has been no discussion here in a while.

**Pull request closed without merging**